extension/src/NABfunctions.ts

```diff
@@ -411,3 +411,47 @@ export async function updateAllXlfFiles() {
 
     console.log('Done: Update all XLF files');
 }
+
+export async function createNewTargetXlf() {
+    console.log("Running: createNewTargetXlf");
+    const targetLanguage: string | undefined = await getUserInput({ placeHolder: "Language code e.g sv-SE" });
+    const selectedMatchBaseApp = await getQuickPickResult(["Yes", "No"], { canPickMany: false, placeHolder: "Match translations from BaseApp?" });
+
+    if (isNullOrUndefined(targetLanguage) || targetLanguage.length === 0) {
+        throw new Error("No target language was set.");
+    }
+    try {
+        const gXlfFile = await WorkspaceFunctions.getGXlfFile();
+        const translationFolderPath = WorkspaceFunctions.getTranslationFolderPath();
+        const matchBaseAppTranslation: boolean = (selectedMatchBaseApp === "Yes");
+        const targetXlfFilepath = path.join(translationFolderPath, `${targetLanguage}.xlf`);
+        if (fs.existsSync(targetXlfFilepath)) {
+            throw new Error(`File already exists: '${targetXlfFilepath}'`);
+        }
+
+        console.log(`Creating new target xlf for language: ${targetLanguage}.\nMatch translations from BaseApp: ${matchBaseAppTranslation}.\nSaving file to path: ${targetXlfFilepath}`);
+        const targetXlfDoc = Xliff.fromFileSync(gXlfFile.fsPath);
+        targetXlfDoc.targetLanguage = targetLanguage;
+        if (matchBaseAppTranslation) {
+            let numberOfMatches = await LanguageFunctions.matchTranslationsFromBaseApp(targetXlfDoc);
+            vscode.window.showInformationMessage(`Added ${numberOfMatches} suggestions from Base Application in ${targetXlfFilepath.replace(/^.*[\\\/]/, '')}.`);
+        }
+        targetXlfDoc.toFileSync(targetXlfFilepath);
```

Comment:
will the new xlf file contain target elements even if user selects "no"?
Maybe "no" should result in a "refresh xlf" without suggestions?

Reply:
Yes, it's created without targets. But running refresh without suggestion would be the same as setting them all as `NotTranslated` right? My first idea was to rework the refreshXlf function to take `Xliff` as parameter instead of `vscode.Uri`,(which at the time of writing this feels like a great idea), although not really necessary when this could be done in `createNewTargetXlf`.

Reply:
refactoring the refreshXlf function to take a `Xliff` parameter sounds as a good idea. Don't know if it's needed in this PR, though.. It might be ok that there are no targets for now?

Reply:
Yes, I think we'll leave it at that for the time being. In my opinion it would be natural to run refresh afterwards either way.